go/vt/tabletserver/gorpcqueryservice/sqlquery.go

```diff
@@ -23,13 +23,40 @@ type SqlQuery struct {
 // GetSessionId is exposing tabletserver.SqlQuery.GetSessionId
 func (sq *SqlQuery) GetSessionId(sessionParams *proto.SessionParams, sessionInfo *proto.SessionInfo) (err error) {
 	defer sq.server.HandlePanic(&err)
-	return sq.server.GetSessionId(sessionParams, sessionInfo)
+	tErr := sq.server.GetSessionId(sessionParams, sessionInfo)
+	tabletserver.AddTabletErrorToSessionInfo(tErr, sessionInfo)
+	if *tabletserver.RPCErrorOnlyInReply {
```

Comment:
If you can solve the StreamExecute case, you do not need this flag.
-> You will not be able to switch this flag without causing compatibility issue (when you do, instead, you can just delete the obsolete code).